Sources/BraveWallet/WalletUserAssetManager.swift

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Data
+import BraveCore
+import Preferences
+
+public protocol WalletUserAssetManagerType: AnyObject {
+  func getAllVisibleAssetsInNetworkAssets(networks: [BraveWallet.NetworkInfo]) -> [NetworkAssets]
+  func getAllUserAssetsInNetworkAssets(networks: [BraveWallet.NetworkInfo]) -> [NetworkAssets]
+  func getUserAsset(_ asset: BraveWallet.BlockchainToken) -> WalletUserAsset?
+  func addUserAsset(_ asset: BraveWallet.BlockchainToken, completion: (() -> Void)?)
+  func removeUserAsset(_ asset: BraveWallet.BlockchainToken, completion: (() -> Void)?)
+  func updateUserAsset(for asset: BraveWallet.BlockchainToken, visible: Bool, completion: (() -> Void)?)
+}
+
+public class WalletUserAssetManager: WalletUserAssetManagerType {
+  
+  private let rpcService: BraveWalletJsonRpcService
+  private let walletService: BraveWalletBraveWalletService
+  
+  public init(
+    rpcService: BraveWalletJsonRpcService,
+    walletService: BraveWalletBraveWalletService
+  ) {
+    self.rpcService = rpcService
+    self.walletService = walletService
+  }
+  
+  public func getAllUserAssetsInNetworkAssets(networks: [BraveWallet.NetworkInfo]) -> [NetworkAssets] {
+    var allVisibleUserAssets: [NetworkAssets] = []
+    for (index, network) in networks.enumerated() {
+      let groupId = "\(network.coin.rawValue).\(network.chainId)"
```

Comment:
This `groupId` is generated as a string in a few spots. Maybe could add a static function on `WalletUserAssetGroup` that takes a `BraveWallet.NetworkInfo` (or just the `chainId` string & `coin` integer if avoiding importing `BraveCore` in the file) and returns the expected `groupId`?

Reply:
i guess i can add a `walletUserAssetGroupId` under `BraveWallet.NetworkInfo` and `BraveWallet.BlockchainToken`

Reply:
added in [5396947](https://github.com/brave/brave-ios/pull/7534/commits/53969478f98cba547e9ae62a8b9edfd75587d7d1)